src/Core/SettingsQuirks.cpp

```diff
@@ -14,19 +15,22 @@
 ///   [2]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=0c54a6a44bf3
 bool nestedEpollWorks(Poco::Logger * log)
 {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0)) && (LINUX_VERSION_CODE < KERNEL_VERSION(5, 6, 13))
-        /// the check is correct since there will be no more 5.5.x releases.
+    if (Poco::Environment::os() != POCO_OS_LINUX)
+        return true;
+
+    DB::VersionNumber linux_version = DB::VersionNumber::fromString(
+        Poco::Environment::osVersion(), /* strict= */ false);
+
+    /// the check is correct since there will be no more 5.5.x releases.
+    if (linux_version >= DB::VersionNumber{5, 5, 0} && linux_version < DB::VersionNumber{5, 6, 13})
```

Comment:
@azat did you check that it will work as expected ? And Poco::Environment::osVersion().
Try to use this in reneameat2.

Reply:
Btw, `uname` and `supportsRenameat2Impl()` do not always work as expected (so `Poco::Environment::osVersion()` and `DB::VersionNumber` will not too, because it works similar way). The problem is that WSL and Docker on Mac pretend to be an ordinary Linux kernel, but actually they are not and behave differently. For example see 
https://github.com/ClickHouse/ClickHouse/blob/549b5f8a0cd037bc9032d976921d64500bfe52e6/src/Common/renameat2.cpp#L66-L75

Reply:
> The problem is that WSL and Docker on Mac pretend to be an ordinary Linux kernel, but actually they are not and behave differently

Right, but there is not a lot, that can be done in this case.

> did you check that it will work as expected ?

Yes.

> Try to use this in reneameat2.

Ok

Reply:
>Try to use this in reneameat2.

Done